### DISCUSSION TITLE
User-defined themes

### DISCUSSION THREAD

**Discussion by gzuuus** [skeletonlabs/skeleton#2103] [Q&A]
I am looking for a way to give the users of my application the possibility to customize their own themes for their user areas. What is the best way to do this? For example how could the skeleton theme generator functionality be developed to preview the theme being generated? ( https://www.skeleton.dev/docs/generator )
I've been reading this article and it certainly has a lot of answers, but I'm missing the glue that binds it all together.
https://www.skeleton.dev/blog/deep-dive-skeleton-styling

Pd I tried to join discord but it looks like the invite has expired.

**Comment by endigo9740** [comment]
Hey @gzuuus, so the problem here is that themes are pre-registered within the Tailwind plugin, which is all sort of constructure and consumed at build time. As far as I'm aware, this cannot be modified at "run time" (when the users are using the app). That's way too late in the process.

So if we talk this out - in theory you would have to do something like this:

1. Provide an interface for users to custom their theme settings (something like our Generator)
2. Generate the theme file itself, upload and implement that into your project source
3. Then rebuild/redeploy your project with the new theme registered in the Tailwind config.

All of this is well beyond the scope of Skeleton itself - and would involve some heavy DevOps knowledge and know how here. But a lot of the business logic for this would happen server-side, so to speak.

Our generator kind of skims this by disabling the registered themes, and instead injecting the generated CSS custom properties of the them itself into the head of the application. It's great for a quick preview - but it's temporary. There's no real persistence of the data here. So not what you're aiming for when talking permanent custom themes.

I'll ping @AdrianGonz97 here, who wrote the v2 Tailwind Plugin and introduced the new theme format to see if he has any other ideas though.

Per Discord. Which link are you using? The one on our website, GitHub, etc? We've had the same invite link for ages, and I've seen folks join as recently as this morning. This should be the one you're using:

https://discord.com/invite/EXqV7W8MtY

**Reply by gzuuus** [reply]
Hello, and first of all thank you for your quick response! 
I've just been looking at the code on the skeleton website and there I've seen how you have developed the theme generator. The stores based approach I think it would work very well for my case, I will follow your approach to develop it in my application. 
Currently the use I want to give it is that users can define their theme settings so that when they visit their profile page it loads with their custom theme. The application I am developing uses nostr as a base layer, a decentralised network, so for persistence I will use nostr, where users will be able to create and save their themes.
